user-manager/src/main/java/com/qpay/usermanager/service/impl/CustomerServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.qpay.usermanager.service.impl;
+
+import com.qpay.usermanager.mapper.CustomerMapper;
+import com.qpay.usermanager.model.dto.customer.CustomerModification;
+import com.qpay.usermanager.model.entity.customer.CustomerEntity;
+import com.qpay.usermanager.repository.CustomerRepository;
+import com.qpay.usermanager.service.CustomerService;
+import com.qpay.usermanager.service.exception.CustomerAlreadyExistsException;
+import com.qpay.usermanager.service.exception.CustomerNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CustomerServiceImpl implements CustomerService {
+
+    private final CustomerRepository customerRepository;
+
+    private final CustomerMapper customerMapper;
+
+    public CustomerEntity getCustomerByEmail(final String email) {
+        final var customerEntity = customerRepository.findCustomerEntityByEmail(email);
+        if (customerEntity == null) {
+            throw new CustomerNotFoundException();
+        }
+        return customerEntity;
+    }
+
+    public CustomerEntity addCustomer(final CustomerModification customerModification) {
+        existsCustomerByEmail(customerModification.email());
```

Comment:
What happen when you try to save a user with duplicated email?

Reply:
it throws CustomerAlreadyExistsException

Reply:
Sorry, I asked incorrectly. 
What happens if you call this method `customerRepository.save(customerEntity);` with an already existing email address?

Reply:
it throws DataIntegrityViolationException, so i decide to process in `try/catch` block and throw my customException

Reply:
Now I don't like how it looks like. Since now if repository throws `DataIntegrityViolationException` you always return Not Found. However, it's not 100% correct,  as this exception can be thrown for many reasons.
You can leave as it is or return more generic exception with this message `e.getMessage()`